src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParser.cs

```diff
@@ -6295,27 +6301,47 @@ internal void TdsLogin(SqlLogin rec, TdsEnums.FeatureExtension requestedFeatures
             _physicalStateObj._messageStatus = 0;
         }// tdsLogin
 
+#if MANAGED_SNI
+        private void SSPIData(byte[] receivedBuff, UInt32 receivedLength, ref byte[] sendBuff, ref UInt32 sendLength)
+        {
+            SNISSPIData(receivedBuff, receivedLength, ref sendBuff, ref sendLength);
+        }
+#else
         private void SSPIData(byte[] receivedBuff, UInt32 receivedLength, byte[] sendBuff, ref UInt32 sendLength)
         {
             SNISSPIData(receivedBuff, receivedLength, sendBuff, ref sendLength);
         }
+#endif
+
 
+#if MANAGED_SNI
+        private void SNISSPIData(byte[] receivedBuff, UInt32 receivedLength, ref byte[] sendBuff, ref UInt32 sendLength)
+#else
         private void SNISSPIData(byte[] receivedBuff, UInt32 receivedLength, byte[] sendBuff, ref UInt32 sendLength)
+#endif
         {
             if (receivedBuff == null)
             {
                 // we do not have SSPI data coming from server, so send over 0's for pointer and length
                 receivedLength = 0;
             }
+
             // we need to respond to the server's message with SSPI data
 #if MANAGED_SNI
-            if (0 != SNIProxy.Singleton.GenSspiClientContext(_physicalStateObj.Handle, receivedBuff, receivedLength, sendBuff, ref sendLength, _sniSpnBuffer, (uint)_sniSpnBuffer.Length)) 
+            uint clientContextResult = SNIProxy.Singleton.GenSspiClientContext(_physicalStateObj.Handle, receivedBuff, ref sendBuff, _sniSpnBuffer);
```

Comment:
Can we change the type of `_sniSpnBuffer` to `string` ? 
We first populate this value in `CreateConnectionHandle` where we convert the string servername to `byte[]` and then we send it down to `GenSspiClientContext` which again converts the _sniSpnBuffer to string to pass it down to Negotiate Stream Pal.